firmware/Core/Src/littlefs/flash_driver.c

```diff
@@ -271,11 +284,16 @@ uint8_t FLASH_write(SPI_HandleTypeDef *hspi, uint8_t chip_number, lfs_block_t ad
     uint8_t status_reg_buffer[1] = {0};
 
     // Keep looping as long as device is busy
-    // FIXME: needs a timeout
+    uint32_t loop_limit = HAL_GetTick() + LOOP_TIMEOUT_MS;
     uint8_t wip = 1;
     uint8_t err = 0;
     while (wip)
     {
+        uint32_t curren_time = HAL_GetTick();
+        if (curren_time > loop_limit) {
+            DEBUG_uart_print_str("Timeout while setting Write Enable Latch to 0\n");
```

Comment:
Change all these error message to be more like: `"Flash write timeout\n"` (that is, explicitly say that a timeout happened, while doing a flash action, and that the flash action attempted was writing). If there's only one place it can timeout while doing a flash write, the detail about it being related to the Write Enable Latch isn't overly helpful for the system as-a-whole.